web-common/src/features/dashboards/state-managers/loaders/get-explore-state-from-session-storage.ts

```diff
@@ -0,0 +1,63 @@
+import { TDDChart } from "@rilldata/web-common/features/dashboards/time-dimension-details/types";
+import { getPartialExploreStateForWebView } from "@rilldata/web-common/features/dashboards/state-managers/loaders/explore-web-view-store";
+import { ExploreUrlWebView } from "@rilldata/web-common/features/dashboards/url-state/mappers";
+import { ExploreStateURLParams } from "@rilldata/web-common/features/dashboards/url-state/url-params";
+import {
+  type V1ExploreSpec,
+  type V1MetricsViewSpec,
+} from "@rilldata/web-common/runtime-client";
+
+/**
+ * Returns explore state filled with extra fields stored when user last visited a particular view.
+ * 1. If no param is set then load the params for the default view from session storage.
+ * 2. If only view param is set then load the params from session storage.
+ * 3. If view=ttd and `measure` is the only other param set load from session storage.
+ */
+export function getExploreStateFromSessionStorage(
+  exploreName: string,
+  storageNamespacePrefix: string | undefined,
+  searchParams: URLSearchParams,
+  metricsViewSpec: V1MetricsViewSpec,
+  exploreSpec: V1ExploreSpec,
+) {
+  if (
+    // exactly one param is set, but it is not `view`
+    (searchParams.size === 1 &&
+      !searchParams.has(ExploreStateURLParams.WebView)) ||
+    // exactly 2 params are set and both `view` and `measure` are not set.
+    (searchParams.size === 2 &&
+      !searchParams.has(ExploreStateURLParams.WebView) &&
+      !searchParams.has(ExploreStateURLParams.ExpandedMeasure)) ||
+    // more than 2 params are set.
+    searchParams.size > 2
+  ) {
+    return undefined;
+  }
+
+  const viewFromUrl = searchParams.get(ExploreStateURLParams.WebView) as
+    | ExploreUrlWebView
+    | undefined;
+  const exploreStateFromSessionStorage = getPartialExploreStateForWebView(
+    exploreName,
+    storageNamespacePrefix,
+    viewFromUrl ?? ExploreUrlWebView.Explore,
+    metricsViewSpec,
+    exploreSpec,
+  );
+  if (!exploreStateFromSessionStorage) {
+    return undefined;
+  }
+
+  if (viewFromUrl === ExploreUrlWebView.TimeDimension) {
+    exploreStateFromSessionStorage.tdd ??= {
+      expandedMeasureName: "",
+      chartType: TDDChart.DEFAULT,
+      pinIndex: -1,
+    };
+    exploreStateFromSessionStorage.tdd.expandedMeasureName = searchParams.get(
+      ExploreStateURLParams.ExpandedMeasure,
+    ) as string;
+  }
```

Comment:
Why do we have special handling for the `TDD` view, but not for the other views?

Reply:
It is because TDD has a parameter - expanded measure. Added some comments to clarify.